src/Parser/Main.ml

```diff
@@ -32,14 +32,15 @@ let rec repl_seq imported () =
 
   | Raw.REPL_Expr e ->
     let def = make_nowhere (Lang.Surface.DReplExpr(Desugar.tr_expr e)) in
-    Seq.Cons(def, repl_seq imported)
+    Seq.Cons([def], repl_seq imported)
 
-  | Raw.REPL_Def def ->
-    Seq.Cons(Desugar.tr_def def, repl_seq imported)
+  | Raw.REPL_Defs defs ->
+      let defs = List.map Desugar.tr_def defs in
+      Seq.Cons(defs, repl_seq imported)
 
   | Raw.REPL_Import import ->
     let imported, defs = Import.import_one imported import in
-    Seq.append (List.to_seq defs) (repl_seq imported) ()
+    Seq.append (List.to_seq (List.map (fun x -> [x]) defs)) (repl_seq imported) ()
```

Comment:
Getting rid of `Seq.append` and changing it to single monolithic `Seq.Cons` hopefully fix #83. If it's a change of single line (that should be modified anyway) we should try.